assets/src/edit-story/components/library/libraryProvider.js

```diff
@@ -49,23 +49,40 @@ function LibraryProvider({ children }) {
   const initialTab = MEDIA.id;
   const [tab, setTab] = useState(initialTab);
   const [textSets, setTextSets] = useState({});
+  const renderedTabs = useRef({});
   const insertElement = useInsertElement();
   const { insertTextSet, insertTextSetByOffset } = useInsertTextSet();
 
   const { showElementsTab } = useFeatures();
 
   // Order here is important, as it denotes the actual visual order of elements.
-  const tabs = useMemo(
-    () => [
+  const tabs = useMemo(() => {
+    return [
       MEDIA,
       MEDIA3P,
-      ...(tab === TEXT.id ? [TEXT] : [{ icon: TextIcon, id: 'text' }]),
+      TEXT,
       SHAPES,
-      ...(showElementsTab ? [ELEMS] : []),
+      showElementsTab ? ELEMS : {},
       PAGE_LAYOUTS,
-    ],
-    [showElementsTab, tab]
-  );
+    ].reduce((accTabs, renderTab) => {
+      // for heavier tab content, avoid rendering the pane unless it is active
+      const isLazyTab = [MEDIA3P.id, TEXT.id, PAGE_LAYOUTS.id].includes(
```

Comment:
💭 nit: Maybe we should make `[MEDIA3P.id, TEXT.id, PAGE_LAYOUTS.id]` a global constant so we can keep a single source of truth and not re-create this every loop? It's not used anywhere else though so it's nbd